TweetMate/src/main/java/com/ybsystem/tweetmate/fragments/dialog/TweetDialog.java

```diff
@@ -0,0 +1,352 @@
+                        R.drawable.ic_hashtag, COLOR_LINK_WEAK, "#" + entity.getText(), "",
+                        null
+                ));
+            }
+        }
+        // User
+        if (menuSetting.contains(USER)) {
+            adapter.add(new Menu(
+                    R.drawable.ic_user, COLOR_STRONG, "@" + mSource.getUser().getScreenName(),
+                    PrefSystem.getProfileThumbByQuality(mSource.getUser()),
+                    () -> ClickUseCase.showUser(mSource.getUser().getId())
+            ));
+            if (mStatus.isRetweet()) {
+                adapter.add(new Menu(
+                        R.drawable.ic_user, COLOR_STRONG, "@" + mStatus.getUser().getScreenName(),
+                        PrefSystem.getProfileThumbByQuality(mStatus.getUser()),
+                        () -> ClickUseCase.showUser(mStatus.getUser().getId())
+                ));
+            }
+            for (TwitterUserMentionEntity entity : mSource.getUserMentionEntities()) {
+                adapter.add(new Menu(
+                        R.drawable.ic_user, COLOR_STRONG, "@" + entity.getScreenName(), "",
+                        () -> ClickUseCase.showUser(entity.getId())
+                ));
+            }
+        }
+        // Detail
+        if (menuSetting.contains(DETAIL)) {
+            adapter.add(new Menu(
+                    R.drawable.ic_doc, COLOR_STRONG, STR_DETAIL, "",
+                    () -> ClickUseCase.showDetail(mSource)
+            ));
+        }
+        // Copy
+        if (menuSetting.contains(COPY)) {
+            adapter.add(new Menu(
+                    R.drawable.ic_copy, COLOR_STRONG, STR_COPY, "",
+                    () -> ClickUseCase.copyText(mSource)
+            ));
+        }
+        // Share
+        if (menuSetting.contains(SHARE)) {
+            adapter.add(new Menu(
+                    R.drawable.ic_share, COLOR_STRONG, STR_SHARE, "",
+                    () -> ClickUseCase.share(mStatus)
+            ));
+        }
+        // OpenTwitter
+        if (menuSetting.contains(TWITTER)) {
+            adapter.add(new Menu(
+                    R.drawable.ic_twitter, COLOR_STRONG, STR_OPEN_TWITTER, "",
+                    () -> ClickUseCase.openStatus(mStatus)
+            ));
+        }
+        // Remove duplication
+        for (int i = 0; i < adapter.getCount(); i++) {
+            for (int j = adapter.getCount() - 1; j > i; j--) {
+                if (adapter.getItem(j).getLabel().equals(adapter.getItem(i).getLabel())) {
+                    adapter.remove(adapter.getItem(j));
+                }
+            }
+        }
+    }
+
+    private void setClickListener(View dialogView, ListView listView, MenuArrayAdapter adapter) {
+        // When tweet clicked
+        dialogView.findViewById(R.id.relative_tweet).setOnClickListener(
+                v -> dismiss()
+        );
+        // When menu item clicked
+        listView.setOnItemClickListener((adapterView, view, i, l) -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1602)